relay-general/src/store/transactions/processor.rs

```diff
@@ -1,21 +1,77 @@
+use chrono::Utc;
+
 use relay_common::SpanStatus;
 
 use crate::processor::{ProcessValue, ProcessingState, Processor};
 use crate::protocol::{
-    Context, ContextInner, Event, EventType, Span, Timestamp, TransactionSource,
+    Context, ContextInner, Event, EventType, Span, Timestamp, TransactionInfo, TransactionSource,
 };
 use crate::store::regexes::TRANSACTION_NAME_NORMALIZER_REGEX;
 use crate::types::{Annotated, Meta, ProcessingAction, ProcessingResult, Remark, RemarkType};
 
+use super::TransactionNameRule;
+
 /// Rejects transactions based on required fields.
 #[derive(Default)]
-pub struct TransactionsProcessor {
+pub struct TransactionsProcessor<'r> {
     normalize_names: bool,
+    tx_name_rules: &'r [TransactionNameRule],
 }
 
-impl TransactionsProcessor {
-    pub fn new(normalize_names: bool) -> Self {
-        Self { normalize_names }
+impl<'r> TransactionsProcessor<'r> {
+    pub fn new(normalize_names: bool, tx_name_rules: &'r [TransactionNameRule]) -> Self {
+        Self {
+            normalize_names,
+            tx_name_rules,
+        }
+    }
+
+    /// Applies the rule if any found to the transaction name.
+    ///
+    /// It find the first rule matching the criteria:
+    /// - source matchining the one provided in the rule sorce, default `url`
+    /// - rule hasn't epired yet
+    /// - glob pattern matches the transaction name
+    pub fn apply_transaction_rename_rule(
+        &self,
+        transaction: &mut Annotated<String>,
+        info: &mut std::option::Option<TransactionInfo>,
+        source: &TransactionSource,
+    ) -> ProcessingResult {
+        let now = Utc::now();
+        transaction.apply(|transaction, meta| {
+            let tran = format!("{}/", transaction.as_str());
```

Comment:
Haven't thought through the consequences, but shouldn't we only add a `/` if it's not already there?

Reply:
Theoretically it should not matter, if there are 1 or 2 slashed at the end. The `**` pattern at the end of each rule will match anything anyway. If I, of course, see it correctly.